services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -563,73 +560,61 @@ public static boolean isValidScope(String scopes, ClientModel client) {
         return true;
     }
 
-    public static Collection<String> parseScopeParameter(String scopeParam) {
-        return new HashSet<>(Arrays.asList(scopeParam.split(" ")));
+    public static Stream<String> parseScopeParameter(String scopeParam) {
+        return Arrays.stream(scopeParam.split(" "));
```

Comment:
As the original collection is a HashSet, wouldn't it be better to use distinct() in the returned stream here?

Reply:
You are right, I forgot to add `distinct()` everywhere.